java/java.source.base/test/unit/src/org/netbeans/api/java/source/TreeUtilitiesTest.java

```diff
@@ -100,6 +102,8 @@ private void prepareTest(String filename, String code, Phase resolutionPhase) th
         
         TestUtilities.copyStringToFile(FileUtil.toFile(testSource), code);
         
+        SourceUtilsTestUtil.setSourceLevel(testSource, sourceLevel);
+
```

Comment:
this will set it to `null` most of the time since `this.sourceLevel` it has no default value. It depends also on what order the tests run, since `testPathForInUnnamedClass` is setting it to `21`, everything before that will use `null` i believe.

(not tested, only looking at it through github)

Reply:
The intent here is not to change anything for existing tests (as we can't be sure the tests would still be testing whatever it was supposed to test). I can wrap the call to `setSourceLevel` with a `if (sourceLevel != null)`, which should make the intent more clear.

Please note that every test method gets its own instance of the class, and hence will start with the `sourceLevel` field initialized to `null`. The test order should not have any effect on the results, I believe.

Reply:
> Please note that every test method gets its own instance of the class,

ok thanks - my mistake. I usually have parameter setup in the `@Before` method (or equivalent) to make it explicit what the non-constant parameters are. But if the whole class resets for every tests case in NB's test setup its essentially doing the same. All good.